### DISCUSSION TITLE
Using sessions

### DISCUSSION THREAD

**Discussion by d-shannon** [osiset/laravel-shopify#1130] [General]
Hey all,

I am using Livewire and auth()->id() is not available after the first interaction with this package. I don't want to set the userId on a public property for obvious reasons. I tried using the laravel session helper, but from what I understand, sessions are being overidden by this package to handle shopify auth. Anyone else encounter this issue?

Cheers

**Comment by ghost** [comment]
`<script type="text/plain" id="shop_name" wire:ignore>{{ \App\Util::getDomainFromRequest() }}</script>`

 
```
public static function getDomainFromRequest(): string
    {
        if (request()->has('host')) {
            return str_replace('/admin', '', base64_decode(request()->get('host')));
        }

        return auth()->user()->name;
    }
```


You will be able to get the domain and log in with every request.

But it's easier to just make user_id public, I don't think anyone really needs it
